src/SourceBuild/content/repo-projects/Directory.Build.targets

```diff
@@ -287,6 +288,76 @@
+
+    https://github.com/dotnet/source-build/issues/3930
+  -->
+  <Target Name="CopyRepoArtifacts"
+          AfterTargets="Package"
+          Condition="Exists($(RepoManifestFile))"
+          Inputs="$(RepoManifestFile)"
+          Outputs="$(BaseIntermediateOutputPath)ArtifactsCopy.complete">
+    <XmlPeek XmlInputPath="$(RepoManifestFile)"
+             Query="Build/Artifact/@Path">
+        <Output TaskParameter="Result" ItemName="RepoManifestArtifact" />
+    </XmlPeek>
+
+    <ItemGroup>
+      <_Asset Include="@(RepoManifestArtifact)" Condition="'%(Extension)' != '.nupkg'"/>
+      <_Package Include="@(RepoManifestArtifact)" Condition="'%(Extension)' == '.nupkg'"/>
+      <_NonShippingPackage Include="@(_Package)" Condition="$([System.String]::Copy('%(Identity)').Contains('$([System.IO.Path]::DirectorySeparatorChar)NonShipping$([System.IO.Path]::DirectorySeparatorChar)'))"/>
+    </ItemGroup>
+
+    <PropertyGroup>
+      <_NupkgsDestination>$(SourceBuiltPackagesPath)</_NupkgsDestination>
+      <_NupkgsDestination Condition="'$(RepositoryName)' == 'source-build-reference-packages'">$(ReferencePackagesDir)</_NupkgsDestination>
+    </PropertyGroup>
+
+    <!-- Building SBRP: At this point the References directory contains the previously-source-built SBRPs,
+         clear it before copying the current SBRPs.  This ensures n-1 SBRPs aren't required to build the product repos. -->
+    <RemoveDir
+      Condition="'$(RepositoryName)' == 'source-build-reference-packages'"
+      Directories="$(ReferencePackagesDir)" />
+
+    <!-- Copy nupkgs -->
+    <Copy
+      Condition="'@(_Package)' != ''"
+      SourceFiles="@(_Package)"
+      DestinationFolder="$(_NupkgsDestination)" />
+
+    <!-- Copy assets -->
+    <Copy
+      Condition="'@(_Asset)' != ''"
+      SourceFiles="@(_Asset)"
+      DestinationFolder="$(SourceBuiltAssetsDir)" />
+
+    <!-- Generate non-shipping package list -->
+    <PropertyGroup>
+      <NonShippingPackagesList Condition="'@(_NonShippingPackage)' != ''">$(PackageListsDir)$(NonShippingPackagesListPrefix)$(RepositoryName).lst</NonShippingPackagesList>
+    </PropertyGroup>
+
+    <WriteLinesToFile
+      Condition="'$(NonShippingPackagesList)' != ''"
+      File="$(NonShippingPackagesList)"
+      Lines="@(_NonShippingPackage->'%(Filename)%(Extension)')"
+      Overwrite="true" />
+
+    <MakeDir Directories="$(BaseIntermediateOutputPath)" />
+    <Touch Files="$(BaseIntermediateOutputPath)ArtifactsCopy.complete" AlwaysCreate="true">
+      <Output TaskParameter="TouchedFiles" ItemName="FileWrites" />
+    </Touch>
+  </Target>
+
+  <!--
+    This target can be removed after source-build picks up version of Arcade that has
+    source-build infra changes for removal od inner-clone in VMR build. Until then the target
+    is needed for building source-build-reference-packages and arcade repos.
+
+    https://github.com/dotnet/source-build/issues/3930
+  -->
   <Target Name="ExtractIntermediatePackages"
```

Comment:
Won't this be needed for source-build-externals since it sill uses the inner clone?

Reply:
Oh yes - I keep conflating the two in my head.